js/theme.js

```diff
@@ -0,0 +1,60 @@
+// system theme check
+if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
+  if(!hasLocalStorage()) {
+    darkModeOn()
+    // remove icon - cannot turn off
+    document.querySelector('#theme-icon-container').remove()
+  } else {
+    // no use local storage til required
```

Comment:
This approach does not utilize local storage at all until the toggle is clicked. I though this would be better. 

The code is messier as a result though. 

It'd be cleaner to just do, but maybe less efficient. 
```
 } else {
    if(isDarkMode === 'true'){
      darkModeOn()
    } else if(isDarkMode === 'false'){
      darkModeOff
    } else {
      darkModeOn()
      localStorage.setItem('darkmode', 'true')
    }

function toggleTheme(e) {
  const isDarkMode = localStorage.getItem('darkmode')
  if (isDarkMode === 'true') {
    localStorage.setItem('darkmode', 'false')
    darkModeOff()
  } else if (isDarkMode === 'false') {
    localStorage.setItem('darkmode', 'true')
    darkModeOn()
}
}
```